src/main/java/com/likelion/backendplus4/talkpick/batch/news/article/infrastructure/collector/writer/ArticleWriter.java

```diff
@@ -44,13 +50,139 @@ public void write(Chunk<? extends List<ArticleEntity>> chunk) {
+	 *
+	 * @param item 처리할 ArticleEntity 객체
+	 */
+	private void processAndSerializeDescription(ArticleEntity item) {
+		String description = item.getDescription();
+		if (description == null || description.isEmpty()) {
+			item.setDescription("[]");
+			return;
+		}
+
+		if (isAlreadyJsonFormat(description)) {
+			return;
+		}
+
+		List<String> paragraphs = splitIntoParagraphs(description);
+		String jsonDescription = serializeToJson(paragraphs);
+		item.setDescription(jsonDescription);
+
+	}
+
+	/**
+	 * 문자열이 JSON 형식인지 확인
+	 *
+	 * @param text 확인할 문자열
+	 * @return JSON 형식이면 true, 아니면 false
+	 */
+	private boolean isAlreadyJsonFormat(String text) {
+		return text.trim().startsWith("[") && text.trim().endsWith("]");
+	}
+
+	/**
+	 * 텍스트를 PARAGRAPH_BREAK를 기준으로 문단으로 분리
+	 *
+	 * @param text 분리할 텍스트
+	 * @return 분리된 문단 리스트
+	 */
+	private List<String> splitIntoParagraphs(String text) {
+		if (!text.contains(PARAGRAPH_BREAK)) {
+			log.warn("PARAGRAPH_BREAK 구분자가 없는 텍스트 감지: {}",
+					text.substring(0, Math.min(text.length(), 50)) + "...");
+			return Arrays.asList(text);
+		}
+
+		String[] paragraphArray = text.split(PARAGRAPH_BREAK);
+		List<String> paragraphs = new ArrayList<>();
+
+		for (String paragraph : paragraphArray) {
+			String trimmed = paragraph.trim();
+			if (!trimmed.isEmpty()) {
+				paragraphs.add(trimmed);
+			}
+		}
+
+		if (paragraphs.isEmpty()) {
+			paragraphs.add(text);
+		}
+
+		return paragraphs;
+	}
+
+	/**
+	 * 문단 리스트를 JSON으로 직렬화한다.
+	 *
+	 * @param paragraphs 직렬화할 문단 리스트
+	 * @return JSON 형식의 문자열
+	 */
+	private String serializeToJson(List<String> paragraphs) {
+		if (paragraphs == null || paragraphs.isEmpty()) {
+			return "[]";
+		}
+
+		StringBuilder sb = new StringBuilder("[");
+		for (int i = 0; i < paragraphs.size(); i++) {
+			String paragraph = paragraphs.get(i);
+			String escaped = escapeJsonString(paragraph);
+			sb.append("\"").append(escaped).append("\"");
+			if (i < paragraphs.size() - 1) {
+				sb.append(",");
+			}
+		}
+		sb.append("]");
+
+		return sb.toString();
+	}
+
+	/**
+	 * JSON 문자열 이스케이프 처리
+	 *
+	 * @param input 이스케이프할 문자열
+	 * @return 이스케이프된 문자열
+	 */
+	private String escapeJsonString(String input) {
+		if (input == null) return "";
+		StringBuilder sb = new StringBuilder();
+		for (int i = 0; i < input.length(); i++) {
+			char c = input.charAt(i);
+			switch (c) {
+				case '\"': sb.append("\\\""); break;
+				case '\\': sb.append("\\\\"); break;
+				case '/': sb.append("\\/"); break;
+				case '\b': sb.append("\\b"); break;
+				case '\f': sb.append("\\f"); break;
+				case '\n': sb.append("\\n"); break;
+				case '\r': sb.append("\\r"); break;
+				case '\t': sb.append("\\t"); break;
+				default:
+					if (c < ' ') {
+						String hex = Integer.toHexString(c);
+						sb.append("\\u");
+						for (int j = 0; j < 4 - hex.length(); j++) {
+							sb.append('0');
+						}
+						sb.append(hex);
+					} else {
+						sb.append(c);
+					}
+			}
+		}
+		return sb.toString();
+	}
```

Comment:
P2:
이 부분은 저도 찾아보니 아래와 같이 리팩토링할 수 있다고 하더라구요 참고하시면 좋을 것 같습니다!

```suggestion
private static final Map<Character, String> JSON_ESCAPES = Map.ofEntries(
    entry('\"', "\\\""),
    entry('\\', "\\\\"),
    entry('/',  "\\/"),
    entry('\b', "\\b"),
    entry('\f', "\\f"),
    entry('\n', "\\n"),
    entry('\r', "\\r"),
    entry('\t', "\\t")
);

protected String escapeJsonString(String input) {
    if (input == null) {
        return "";
    }

    return input.chars()
        .mapToObj(cp -> {
            char c = (char) cp;
            if (JSON_ESCAPES.containsKey(c)) {
                return JSON_ESCAPES.get(c);
            }
            if (cp < 0x20) {
                return String.format("\\u%04x", cp);
            }
            return String.valueOf(c);
        })
        .collect(Collectors.joining());
}

```

Reply:
기존의 for, switch문 대신 스트림 api를 사용한 방식은 좋은거같습니다.
JSON_ESCAPES를 상수로 빼서 가독성도 좋고 나중에 escape를 추가하기도 편하겠네요